crates/core/tedge_agent/src/agent.rs

```diff
@@ -267,12 +268,23 @@ pub struct SmAgent {
     config: SmAgentConfig,
     operation_logs: OperationLogs,
     persistence_store: AgentStateRepository,
-    _flock: Flockfile,
+    _flock: Option<Flockfile>,
 }
 
 impl SmAgent {
     pub fn try_new(name: &str, mut config: SmAgentConfig) -> Result<Self, AgentError> {
-        let flock = check_another_instance_is_not_running(name, &config.run_dir)?;
+        let config_repository =
+            tedge_config::TEdgeConfigRepository::new(config.config_location.clone());
+        let tedge_config = config_repository.load()?;
```

Comment:
__Minor__. Since the config is already read to populate the `SmAgentConfig`, I wonder if one can simply extend the `SmAgentConfig` struct with a `use_lock` property that would be used here.

Reply:
It makes sense. I extended `SmAgentConfig` and now it looks for this flag there.